### ISSUE TITLE
ports closed : caldav, zarafa mapi

### ISSUE THREAD

**Issue by osnet** [mlan/docker-kopano#3]
hi,
there is some strange behavior : ) 
- The webapp is running fine with :80 
- zpush / eas passed via proxy on :80
- opened / exposed ports  8080 and 236 (testing)

## DOCKER info

````
[root@vsrv-mail-02 kopano]# docker --version
Docker version 19.03.12, build 48a66213fe

[root@vsrv-mail-02 kopano]# docker info
Client:
 Debug Mode: false

Server:
 Containers: 3
  Running: 3
  Paused: 0
  Stopped: 0
 Images: 3
 Server Version: 19.03.12
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local local-persist
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1127.13.1.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 3.675GiB
 Name: vsrv-mail-02
 ID: S5PM:P7W5:ZRAU:ABE2:OEPK:VS24:LW5A:AXAM:H3Q7:FGCL:4HX3:GNAL
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false


````

docker ps : 
```
94f46c5a2539        mlan/kopano:full      "docker-entrypoint.s…"   3 minutes ago       Up 3 minutes (healthy)   192.168.220.4:236->236/tcp, 192.168.220.4:443->443/tcp, 0.0.0.0:80->80/tcp, 0.0.0.0:8080->8080/tcp, 192.168.220.4:2003->2003/tcp   test_app_1
```

docker container port

```
[root@vsrv-mail-02 kopano]# docker container port 94f46c5a2539
8080/tcp -> 0.0.0.0:8080
2003/tcp -> 192.168.220.4:2003
236/tcp -> 192.168.220.4:236
443/tcp -> 192.168.220.4:443
80/tcp -> 0.0.0.0:80

```

netstat .... 

```
[root@vsrv-mail-02 kopano]# netstat -anp | grep :
tcp        0      0 192.168.220.4:236       0.0.0.0:*               LISTEN      108912/docker-proxy 
tcp        0      0 192.168.220.4:143       0.0.0.0:*               LISTEN      108307/docker-proxy 
tcp        0      0 192.168.220.4:2003      0.0.0.0:*               LISTEN      108694/docker-proxy 
tcp        0      0 192.168.220.4:25        0.0.0.0:*               LISTEN      108327/docker-proxy 
tcp        0      0 192.168.220.4:443       0.0.0.0:*               LISTEN      108785/docker-proxy 
tcp6       0      0 :::80                   :::*                    LISTEN      109015/docker-proxy 
tcp6       0      0 :::8080                 :::*                    LISTEN      108610/docker-proxy 
```

firewall

```
[root@vsrv-mail-02 kopano]# firewall-cmd --list-ports
236/tcp 443/tcp 143/tcp 8080/tcp

```

container internal:

```
root@94f46c5a2539:/# curl -vvv http://127.0.0.1:8080
* Rebuilt URL to: http://127.0.0.1:8080/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.58.0
> Accept: */*
> 
< HTTP/1.1 401 Unauthorized
< WWW-Authenticate: Basic realm="Kopano CalDav Gateway"
< Content-Length: 0
< Server: Kopano
< Date: Wed, 28 Oct 2020 09:11:28 GMT
< Connection: close
< 
* Closing connection 0
```

external 
reject

```
[root@vsrv-mail-02 kopano]# curl -vvv http://127.0.0.1:8080
* About to connect() to 127.0.0.1 port 8080 (#0)
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: 127.0.0.1:8080
> Accept: */*
> 
* Recv failure: Die Verbindung wurde vom Kommunikationspartner zurückgesetzt
* Closing connection 0
curl: (56) Recv failure: Die Verbindung wurde vom Kommunikationspartner zurückgesetzt

```

8080 and 236 ..  there is no damn was to make them accessable from the outside ... 
any thoughts on this ? : ))

**Comment by mlan** [comment]
Hi @osnet 

Perhaps you can share relevant parts of your `docker-compose.yml` file to shed some light on the port mappings was set up?
 
It would also be very useful if you can also share the relevant parts of the kopano config files you used for caldav and mapi.

Admittedly, I have not tested caldav that much, but I am willing to do that in the coming days if you help me understand you use-case better.

**Comment by osnet** [comment]
ARGH  sometimes i do not see the goddamn trees while being in the forest 
ical is fixed .... 
i did not think ok adding the ical.cfg ... fml :facepalm:  : ) 
caldav / ical is running fine with evolution ... 

this is my added ical.cfg

```
# See the kopano-ical.cfg(5) manpage for details and more directives.

# Space-separated list of address:port specifiers with optional %interface
# infix for where the server should listen for connections.
# ical has often been placed on 8080 and icals on 8443.
#
ical_listen = 0.0.0.0:8080
#icals_listen =
#tls_min_proto = tls1.2
# File with RSA key for SSL
#ssl_private_key_file = /etc/kopano/ical/privkey.pem
# File with certificate for SSL
#ssl_certificate_file = /etc/kopano/ical/cert.pem

# Verify client certificate
#ssl_verify_client = no
# Client verify file and/or path
#ssl_verify_file =
#ssl_verify_path =

# default connection to the storage server
# Please refer to the administrator manual or manpage why HTTP is used rather than the UNIX socket.
server_socket = http://localhost:236/

#log_method = auto
# Loglevel (0(none), 1(crit), 2(err), 3(warn), 4(notice), 5(info), 6(debug))
log_level = 3
log_file = /var/log/kopano/ical.log
log_timestamp = yes

# The timezone of the system clock
server_timezone = Europe/Berlin
# Enable the iCalendar GET method for downloading calendars
enable_ical_get = yes
```

forgive me : ))) 
i am now checking why mapi is not working as expected ^^

**Comment by mlan** [comment]
Great news! :)

I will consider adding support for ical configuration via envvars in the near future.

**Comment by mlan** [comment]
Fixed in 2acaf05d930667422ef88030c00f9d07d8f3005d. See [Enabling IMAP, POP3 and ICAL](https://github.com/mlan/docker-kopano#enabling-imap-pop3-and-ical). Mark as closed.
